chirp/drivers/tk690.py

```diff
@@ -1043,9 +1106,20 @@ def set_settings(self, settings):
             for button in group:
                 groupKey = group.get_name()
                 settingKey = button.get_name()
-                self._memobj[groupKey][settingKey] = \
-                    [value for (key, value) in BUTTON_FUNCTION_LIST
-                        if key == button.value.get_value()][0]
+                if groupKey == "button_assignments":
+                    self._memobj[groupKey][settingKey] = \
+                        [value for (key, value) in BUTTON_FUNCTION_LIST
+                            if key == button.value.get_value()][0]
+                elif groupKey == "basic_settings":
+                    if settingKey == "head_type":
+                        self._memobj.properties.rid.headtype = \
+                            button.value.__int__()
```

Comment:
Erm, this should be `int(button.value)` here (and below) right? You shouldn't be calling the dunder methods directly. Perhaps left over from debugging the lack of refresh thing?